python-frontend/src/main/java/org/sonar/python/semantic/SymbolUtils.java

```diff
@@ -230,34 +230,32 @@ public static int firstParameterOffset(FunctionSymbol functionSymbol, boolean is
     return 0;
   }
 
-  public static Optional<FunctionSymbol> getOverriddenMethod(FunctionSymbol functionSymbol) {
+  public static List<FunctionSymbol> getOverriddenMethods(FunctionSymbol functionSymbol) {
```

Comment:
While the change in behavior makes sense for what we need, I think we still want to keep the original behavior accessible.

What I had in mind was to keep the methods
```
public static Optional<FunctionSymbol> getOverriddenMethod(FunctionSymbol functionSymbol)
Optional<FunctionSymbol> getFunctionSymbol(@Nullable Symbol symbol)
```
and then have methods that would do:
```
public static Optional<FunctionSymbol> getOverriddenMethod(FunctionSymbol functionSymbol, Predicate<AmbiguousSymbol> disambiguationPredicate)
Optional<FunctionSymbol> getFunctionSymbol(@Nullable Symbol symbol, Predicate<AmbiguousSymbol> disambiguationPredicate)
```

With the default predicate being just something that always returns false and then a `isSameArguments` that would behave as we expect here.

So that by default the basic methods always return a single value iff there is only one, and the callee doesn't have to call anything else.